### ISSUE TITLE
Incorrect documentation (and possible work) of `remark-lint-list-item-indent`

### ISSUE THREAD

**Issue by AlexWayfer** [remarkjs/remark-lint#223]
### Subject of the issue

I see here: https://github.com/remarkjs/remark-lint/tree/57b9eb1/packages/remark-lint-list-item-indent

> Options: `'tab-size'`, `'mixed'`, or `'space'`, default: `'tab-size'`.
> ...
> This can be configured with the [`listItemIndent`](https://github.com/remarkjs/remark/tree/master/packages/remark-stringify#optionslistitemindent) option.

And here: https://github.com/remarkjs/remark/tree/master/packages/remark-stringify#optionslistitemindent

> **`options.listItemIndent`**
>
> Style of indentation for list items (`'tab'`, `'mixed'` or `'1'`, default: `'tab'`).
>
> * `'tab'`: use a tab stops (4 spaces)
> * `'1'`: use one space
> * `'mixed'`: use 1 for tight and tab for loose list items

So, should I use `tab` or `tab-size`, `space` or `1`?

I'm trying to use `space` and getting errors, `1` — and it does not affect.

### Your environment

*   **OS**: Arch Linux
*   **Packages**: `remark: 11.0.2, remark-cli: 7.0.1`
*   **Env**: `node v13.7.0`

### Steps to reproduce

```yaml
# .remarkrc.yaml
plugins:
  - remark-lint-list-item-indent

settings:
  listItemIndent: '1'
```

(`README.md` is from `valid` example)

```
> remark README.md
- List item.

Paragraph.

11. List item

Paragraph.

- List
  item.

- List
  item.
README.md
   1:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
   9:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint
  12:3  warning  Incorrect list-item indent: add 2 spaces  list-item-indent  remark-lint

⚠ 3 warnings
```

### Expected behaviour

Identical documentation and no warnings after configuration.

### Actual behaviour

I don't sure about documentation and `listItemIndent: '1'` changes nothing for me.

**Comment by wooorm** [comment]
This part is for the lint rule, which produces warnings (or not):

```yaml
plugins:
  - remark-lint-list-item-indent
```

You can configure the lint rule to prefer a single space like so:

```yaml
plugins:
  - - remark-lint-list-item-indent
    - space

```

---

This part is for remark. It configures how Markdown is parsed and serialised.
With your setup, if you are serialising markdown, a single space is always used:

```yaml
settings:
  listItemIndent: '1'
```

**Comment by AlexWayfer** [comment]
Oh… not so obvious… okay, thank you! Sorry.

**Comment by AlexWayfer** [comment]
But… can I somehow to configure this (`lint-list-item-indent`) to be an error instead of warning? 

```yaml
plugins:
  - - remark-lint-list-item-indent
    - space
    - error
```

doesn't work. The status of `remark` (CLI) is `0`. `-error; - space` doesn't valid at all.

Oh, I found it. So unintuitive… I need to "go deeper" (nest deeper), and in specific order (which was invalid in flat structure, ouch):

```yaml
plugins:
  - - remark-lint-list-item-indent
    - - error
      - space
```

**Comment by wooorm** [comment]
With the CLI, you can add `--frail` to fail on warnings. See `remark --help` for more info.

Alternative, indeed, what you found ;)
I personally don’t use yaml, and now I see it, yaml is pretty verbose lol

**Comment by AlexWayfer** [comment]
But… what if I want to use `preset-lint-recommended` and configure `lint-list-item-indent`?… Oh… such pain. :grinning: 

```yaml
plugins:
  - - remark-preset-lint-recommended
  - - remark-lint-list-item-indent
    - - error
      - space
```

gives:

```
README.md
  1:1  error  TypeError: value.concat is not a function
    at coerce (/home/alex/node_modules/unified-lint-rule/index.js:82:20)
    at Function.remark-lint:list-item-indent (/home/alex/node_modules/unified-lint-rule/index.js:24:18)
    at freeze (/home/alex/.config/yarn/global/node_modules/unified/index.js:118:28)
    at Function.parse (/home/alex/.config/yarn/global/node_modules/unified/index.js:266:5)
    at parse (/home/alex/.config/yarn/global/node_modules/unified-engine/lib/file-pipeline/parse.js:43:36)
    at wrapped (/home/alex/.config/yarn/global/node_modules/trough/wrap.js:25:19)
    at next (/home/alex/.config/yarn/global/node_modules/trough/index.js:57:24)
    at done (/home/alex/.config/yarn/global/node_modules/trough/wrap.js:55:16)
    at handleConfiguration (/home/alex/.config/yarn/global/node_modules/unified-engine/lib/file-pipeline/configure.js:68:5)
    at done (/home/alex/.config/yarn/global/node_modules/unified-engine/lib/configuration.js:75:14)
```

**Comment by AlexWayfer** [comment]
OK, if I'd get the conf like there: https://github.com/remarkjs/remark-lint/blob/b539b24/readme.md#configuring-remark-lint

It'd be like:

```yaml
plugins:
  - remark-preset-lint-recommended
  - - remark-lint-list-item-indent
    - - error
      - space
```

And the same error.

**Comment by AlexWayfer** [comment]
OK, it looks like a bug:

```yaml
plugins:
  - remark-preset-lint-recommended
  - - remark-lint-list-item-indent
    - false # or `tab-size`, or `space`, etc.
```

works, and adding `- error` with nesting (like above) — does not.

**Comment by wooorm** [comment]
Ahh, pfft, that’s difficult.

In the preset, the rule is defined as having a “value” of a string I think?
Then you are redefining it as an array.
Which caused the exception.

...Now the question is, how to merge a string with an array?

**Comment by AlexWayfer** [comment]
> In the preset, the rule is defined as having a “value” of a string I think?

Which specifically? But yes, `false` in YAML is `false`, `space` is `'space'`.

This YAML could be converted to JSON:

```json
{
  "plugins": [
    "remark-preset-lint-recommended",
    ["remark-lint-list-item-indent", ["error", "space"]]
  ]
}
```

So, I can do:

```json
{
  "plugins": [
    "remark-lint-list-item-indent", ["error", "space"]
  ]
}
```

and I can do:

```json
{
  "plugins": [
    "remark-preset-lint-recommended",
    ["remark-lint-list-item-indent", "space"]
  ]
}
```

but I can't combine them, configure one of plugins with error-level.

**Comment by wooorm** [comment]
Here a string is defined:

https://github.com/remarkjs/remark-lint/blob/b539b24f7bb42d915d991b6ba0304ec0b2f8f880/packages/remark-preset-lint-recommended/index.js#L15

Here the extension happens and fails:

https://github.com/unifiedjs/unified/blob/952e15d10301dea95c43001460bc52992729da32/index.js#L235-L237

**Comment by AlexWayfer** [comment]
So… I should go and create an issue for `unified`, right?

**Comment by wooorm** [comment]
Im already working on that, thanks!

**Comment by AlexWayfer** [comment]
> Im already working on that, thanks!

Great, thank you! I appreciate it.
